python/GafferImageTest/CatalogueTest.py

```diff
@@ -66,6 +67,16 @@ def sendImage( image, catalogue, extraParameters = {}, waitForSave = True, close
 
 	__catalogueIsRenderingMetadataKey = "gaffer:isRendering"
 
+	def tearDown( self ) :
+
+		for root, dirs, files in os.walk( self.temporaryDirectory() ) :
+			for fileName in files :
+				( pathlib.Path( root ) / fileName ).chmod( stat.S_IRWXU )
+			for dirName in dirs :
+				( pathlib.Path( root ) / dirName ).chmod( stat.S_IRWXU )
```

Comment:
Can we move this to `GafferTest.TestCase.tearDown()`? I think that would mean we could remove the following :

- ImageWriterTest.tearDown
- ApplicationRootTest.testPreferencesPermissionsErrors cleanup
- ScriptNodeTest.testReadOnlyMetadata cleanup
- Future cleanup in IECoreArnoldTest.RendererTest